### ISSUE TITLE
RL Agent Implementation (DQN)

### ISSUE THREAD

**Issue by johnhnguyen97** [Nguyenetic/argus#2]
## Overview

Implement Deep Q-Network (DQN) reinforcement learning agent for adaptive bot evasion.

## Goals
- Learn optimal scraping strategies to evade bot detection
- Adapt behavior based on website responses
- Achieve >80% evasion rate against common anti-bot systems

### State Space
- Page load time
- CAPTCHA presence
- Rate limit signals (429, 403 responses)
- JavaScript challenge detection
- Cloudflare/DataDome detection

### Action Space
- Request timing (delays between requests)
- Mouse movement patterns
- Scrolling behavior
- User-agent rotation
- Cookie handling strategy

### Reward Function
- +10: Successful page scrape
- -10: CAPTCHA triggered
- -5: Rate limited
- -20: IP banned
- +5: Stealth mode undetected

### Core DQN
- [ ] Implement Q-Network with `tch-rs` (PyTorch bindings)
- [ ] Design neural network architecture (3-4 layers)
- [ ] Implement experience replay buffer (10K capacity)
- [ ] Implement epsilon-greedy exploration (ε=0.1)
- [ ] Target network with periodic updates

### Training Environment
- [ ] Create synthetic bot detection simulator
- [ ] Implement reward calculation
- [ ] Add episode tracking and logging
- [ ] Save/load trained models
- [ ] Visualize training progress

### Integration
- [ ] Integrate with browser automation
- [ ] Add CLI flag `--rl-mode`
- [ ] Implement real-time action selection
- [ ] Add agent performance metrics

## Example Usage
```bash
# Train agent
argus train-rl --episodes 1000 --save-path models/dqn.pt

# Use trained agent
argus scrape https://example.com --rl-mode --model models/dqn.pt
```

## Performance Targets
- Training time: <2 hours on CPU
- Inference time: <1ms per action
- Evasion rate: >80% (MVP), >98% (final)
- Memory usage: <500MB

## References
- Week 3 in DETAILED_ROADMAP_RUST.md
- Research: Rainbow DQN paper
- `tch-rs` docs: https://docs.rs/tch

**Phase:** 1 - Foundation (Week 3)
**Priority:** High
